src/js/apps/patients/schedule/schedule_app.js

```diff
@@ -0,0 +1,78 @@
+import { clone, reduce, keys } from 'underscore';
+import Backbone from 'backbone';
+import Radio from 'backbone.radio';
+
+import App from 'js/base/app';
+
+import StateModel from './schedule_state';
+
+import FiltersApp from './filters_app';
+
+import { LayoutView, ScheduleTitleView, TableHeaderView } from 'js/views/patients/schedule/schedule_views';
+
+export default App.extend({
+  StateModel,
+  childApps: {
+    filters: {
+      AppClass: FiltersApp,
+      regionName: 'filters',
+      restartWithParent: false,
+      getOptions: ['currentClinician'],
+    },
+  },
+  onBeforeStart() {
+    this.showView(new LayoutView({
+      state: this.getState(),
+    }));
+
+    this.getRegion('list').startPreloader();
+
+    this.showScheduleTitle();
+    this.startFiltersApp();
+    this.showTableHeaders();
+  },
+  beforeStart() {
+    const filter = this.getState().getEntityFilter();
+    return Radio.request('entities', 'fetch:actions:collection', { filter });
+  },
+  onStart(options, collection) {
+    this.collection = collection;
+  },
+  showScheduleTitle() {
+    const filters = this.getState().getFilters();
+    const owner = Radio.request('entities', 'clinicians:model', filters.clinicianId);
+    const role = Radio.request('entities', 'roles:model', filters.roleId);
+
+    this.showChildView('title', new ScheduleTitleView({
+      owner,
+      role,
+    }));
+  },
+  startFiltersApp() {
+    const filtersApp = this.startChildApp('filters', {
+      state: this.getState().getFilters(),
+      shouldShowClinician: this.getState().id !== 'shared-by',
+      shouldShowRole: this.getState().id !== 'owned-by',
+      shouldShowDate: this.getState().id === 'shared-by' || this.getState().id === 'owned-by',
+    });
+
+    this.listenTo(filtersApp.getState(), 'change', ({ attributes }) => {
+      this.setState({ filters: clone(attributes) });
+    });
+  },
+  showTableHeaders() {
+    const tableHeadersView = new TableHeaderView();
+
+    this.showChildView('table', tableHeadersView);
+  },
+  _getScheduledByDate() {
+    const groupedCollection = this.collection.groupBy('due_date');
```

Comment:
This is what I'm thinking for formatting the original action collection into one grouped by date for the collectionView to display. We can use buildChildView to send actions to a child collection view as a collection to display actions by date.